src/create-element.js

```diff
@@ -100,3 +79,32 @@ export function Fragment(props) {
  */
 export const isValidElement = vnode =>
 	vnode != null && vnode.constructor === undefined;
+
+export function createBackingNode(vnode) {
+	const backingNode = {
+		key: vnode.props ? vnode.props.key : undefined,
+		ref: vnode.props ? vnode.props.ref : undefined,
+		type: vnode.type,
+		_node: vnode,
+		_children: null,
+		_parent: null,
+		_depth: 0,
+		_dom: null,
+		// _nextDom must be initialized to undefined b/c it will eventually
+		// be set to dom.nextSibling which can return `null` and it is important
+		// to be able to distinguish between an uninitialized _nextDom and
+		// a _nextDom that has been set to `null`
+		_nextDom: undefined,
+		_component: null
+	};
+
+	if (vnode.props && vnode.props.ref) {
+		delete vnode.props.ref;
```

Comment:
It's likely much faster to loop over the props and copy them. Does that end up being much larger?

One perk of the loop is that we don't have to have a megamorhpic property access for `props.ref` and `props.key` - they can be local variables that get assigned to if the loop encounters `key` or `ref` keys.